pkg/controllers/disruption/multinodeconsolidation.go

```diff
@@ -120,42 +122,50 @@ func (m *MultiNodeConsolidation) firstNConsolidationOption(ctx context.Context,
 	lastSavedCommand := Command{}
 	lastSavedResults := scheduling.Results{}
 	// Set a timeout
-	timeout := m.clock.Now().Add(MultiNodeConsolidationTimeoutDuration)
-	// binary search to find the maximum number of NodeClaims we can terminate
+	timeoutCtx, cancel := context.WithDeadline(ctx, m.clock.Now().Add(MultiNodeConsolidationTimeoutDuration))
+	defer cancel()
 	for min <= max {
-		if m.clock.Now().After(timeout) {
+		// Check for timeout using select
+		select {
```

Comment:
Can we just check for `if ctx.Err() != nil` here, and does that make this a bit cleaner? (no select effectively and a smaller diff)